subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/statistics/TaskExecutionStatisticsEventAdapter.java

```diff
@@ -49,10 +50,21 @@ public void beforeExecute(Task task) {
     public void afterExecute(Task task, TaskState state) {
         if (!taskIsForNestedBuild(task)) {
             TaskStateInternal stateInternal = (TaskStateInternal) state;
-            if (stateInternal.isAvoided()) {
-                avoidedTasksCount++;
-            } else if (stateInternal.isActionsWereExecuted()) {
-                executedTasksCount++;
+            if (stateInternal.isActionable()) {
+                switch (stateInternal.getOutcome()) {
```

Comment:
:o: nitpick, part of the changes in 4.0 were intended to prevent other areas of code from having logic using TaskOutcomes. The idea being that we want to change the outcomes hierarchy, so we want as much logic as possible to stay in the TaskState. 

Suppose we introduce a new outcome `EXECUTED_INCREMENTALLY`, then this code would silently break.